corona-warn-app/cwa-app-android
# Discard OTP when high risk card disappears (EXPOSUREAPP-5144)

Fulfills the requirement from the Survey story about: 

> When the high-risk card disappears (status change red --> green), a previously generated OTP is discarded.

There are already some unit tests in `RiskLevelChangeDetectorTest`

How to test it manually: 
- Trigger high-risk card (Test Menu => ENF v2 Calculation => INCREASED_RISK_DEFAULT => Calculate Risk Level)
- See high-risk card in HomeMenu; Enter the Risk details screen to see the survey (should currently work on INT)
- Trigger low-risk card (Test Menu => ENF v2 Calculation => LOW_RISK_DEFAULT => Calculate Risk Level)
- Check LogCat => "Discarding one time password for survey about previous high-risk state." should be printed out
